### ISSUE TITLE
Ignore null values in $sql array?

### ISSUE THREAD

**Issue by J-roen** [doctrine/migrations#837]
Bugs in Doctrine DBAL, like https://github.com/doctrine/dbal/issues/3621, can result in null values in the `$sql` array in `\Doctrine\Migrations\Generator\SqlGenerator::generate()`.

In version 1.* this results in a `$this->addSql(NULL);` statement in the generated migration.

In version 2.* this results in the error `stripos() expects parameter 1 to be string, null given` at `/app/vendor/doctrine/migrations/lib/Doctrine/Migrations/Generator/SqlGenerator.php:48`.

Of course the best solution is to fix these bugs in Doctrine DBAL. Would it however be an idea to also ignore these null values in `\Doctrine\Migrations\Generator\SqlGenerator::generate()`?

**Comment by SenseException** [comment]
I agree that it should be fixed in DBAL, but as long as older versions are valid dependencies, `null` values have to be checked IMHO here too to prevent type errors on strict type mode.
